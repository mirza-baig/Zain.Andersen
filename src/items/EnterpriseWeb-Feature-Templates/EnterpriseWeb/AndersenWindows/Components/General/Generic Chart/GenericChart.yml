---
ID: "490104e2-2fe9-4b2c-8074-8bb42b86a1f7"
Parent: "563c8f98-28ba-4fa0-ac12-c50022edf84a"
Template: "ab86861a-6030-46c5-b394-e8f99e8b87db"
Path: /sitecore/templates/Feature/EnterpriseWeb/AndersenWindows/Components/General/Generic Chart/GenericChart
SharedFields:
- ID: "06d5295c-ed2f-4a54-9bf2-26228d113318"
  Hint: __Icon
  Value: "Business/32x32/column-chart.png"
- ID: "12c33f3f-86c5-43a5-aeb4-5598cec45116"
  Hint: __Base template
  Value: |
    {B61A1D20-9EE5-4CCB-9E0D-24A97E58D293}
    {99969A89-B62F-4155-A052-0F54C8C21B52}
    {5E2F3CF0-45EC-4618-AA30-0465290A248D}
- ID: "b03569b1-1534-43f2-8c83-bd064b7d782c"
  Hint: __Renderers
  Value: 
- ID: "f7d48a55-2158-4f02-9356-756654404f73"
  Hint: __Standard values
  Value: "{91DF3A92-4DC5-42A6-BCF6-015B0424E50F}"
Languages:
- Language: en
  Fields:
  - ID: "b5e02ad9-d56f-4c41-a065-a133db87bdeb"
    Hint: __Display name
    Value: Generic Chart
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20231211T232821Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\jhuynh
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\jhuynh
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "cc741640-9b64-4671-8799-a97b900d3ac4"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\jhuynh
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20231212T222652Z
