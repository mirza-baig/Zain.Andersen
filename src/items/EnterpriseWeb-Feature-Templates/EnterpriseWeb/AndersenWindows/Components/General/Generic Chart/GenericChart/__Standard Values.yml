---
ID: "91df3a92-4dc5-42a6-bcf6-015b0424e50f"
Parent: "490104e2-2fe9-4b2c-8074-8bb42b86a1f7"
Template: "490104e2-2fe9-4b2c-8074-8bb42b86a1f7"
Path: /sitecore/templates/Feature/EnterpriseWeb/AndersenWindows/Components/General/Generic Chart/GenericChart/__Standard Values
SharedFields:
- ID: "1172f251-dad4-4efb-a329-0c63500e4f1e"
  Hint: __Masters
  Value: "{19DADBBA-95AC-4F1E-867A-C64D778B90C6}"
- ID: "b03569b1-1534-43f2-8c83-bd064b7d782c"
  Hint: __Renderers
  Value: "{120AD31C-0E1E-41B5-881D-D5C2916B4E90}"
Languages:
- Language: en
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20231212T001322Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\jhuynh
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\jhuynh
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "df9772ce-ceec-4b51-8ef2-edb8fb39f8ee"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\jhuynh
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20231212T222713Z
