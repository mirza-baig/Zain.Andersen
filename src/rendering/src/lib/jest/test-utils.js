// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
