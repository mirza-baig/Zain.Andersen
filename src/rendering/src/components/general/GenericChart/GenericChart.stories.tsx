// Global
import { Story, Meta } from '@storybook/react';
// Local
import GenericChart, { GenericChartProps } from './GenericChart';
import defaultData from './GenericChart.mock-data';

export default {
  title: 'General/GenericChart',
  component: GenericChart,
} as Meta;
const Template: Story<GenericChartProps> = (props) => <GenericChart {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
