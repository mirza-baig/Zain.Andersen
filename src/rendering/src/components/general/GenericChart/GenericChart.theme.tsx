// Lib
import { ThemeFile } from 'lib/context/ThemeContext';
import { BackgroundColor, TextAlignment } from './GenericChart';

const GetBackgroundColorStyling = (backgroundColor: BackgroundColor) => {
  switch (backgroundColor) {
    case 'black':
      return 'bg-black text-white';
    case 'gray':
      return 'bg-gray text-black';
    case 'white':
    default:
      return 'bg-white text-black';
  }
};

const GetBorderColorStyling = (backgroundColor: BackgroundColor) => {
  switch (backgroundColor) {
    case 'black':
      return 'border-b-gray border-b';
    case 'gray':
    case 'white':
    default:
      return 'border-b-black border-b';
  }
};

export const GetTextAlignment = (textAlignment: TextAlignment) => {
  switch (textAlignment) {
    case 'center':
      return 'text-center';
    case 'right':
      return 'text-right';
    case 'left':
      return 'text-left';
    default:
      return '';
  }
};

export const GenericChartTheme = (
  titleRowBackground: BackgroundColor,
  titleColumnBackground: BackgroundColor,
  textAlignment: TextAlignment
): ThemeFile => {
  return {
    aw: {
      classes: {
        section: `font-futura-pt`,
        table: ` w-full table-fixed ${GetTextAlignment(textAlignment)}`,
        theadRow: ` title-row hidden md:table-row `,
        tableRow: ` content-row `,
        titleRow: ` border-b-black ${GetBackgroundColorStyling(titleRowBackground)}`,
        imageRow: ` image-row hidden md:table-row`,
        titleColumn: `  ${GetBackgroundColorStyling(titleColumnBackground)}`,
        columnHeader: `column-header border border-black ${GetBackgroundColorStyling(
          titleColumnBackground
        )} ${GetBorderColorStyling(
          titleColumnBackground
        )} text-[14px] md:text-[18px] p-[5px_15px] font-demi`,
        tableData: `bg-white border border-black block md:table-cell border-b-black`,
        rowHeader: `row-header p-[5px] ${GetBackgroundColorStyling(
          titleRowBackground
        )} leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h3]:mb-[10px] [&_h4]:text-[18px] [&_h4]:font-demi [&_h4]:mb-[8px]`,
        mobileTitleColumn: ` mobile-title md:hidden p-[5px] ${GetBackgroundColorStyling(
          titleRowBackground
        )} ${GetBorderColorStyling(
          titleRowBackground
        )} leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h4]:text-[14px] [&_h4]:font-demi`,
        contentClass: ` [&_img]:mx-auto p-[5px]`,
        imaageColumnHeader: `image-col`,
        imageColumn: `image-col p-[5px] max-md:hidden`,
        footer: `content-footer`,
      },
    },
    rba: {
      classes: {},
    },
  };
};
