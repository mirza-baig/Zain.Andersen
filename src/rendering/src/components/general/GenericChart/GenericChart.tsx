// Global
import {
  ImageField,
  LinkField,
  RichTextField,
  Text,
  TextField,
  withDatasourceCheck,
} from '@sitecore-jss/sitecore-jss-nextjs';
import { useTheme } from 'lib/context/ThemeContext';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';

// Components
import { BodyCopy } from 'src/helpers/BodyCopy';
import { Component } from 'src/helpers/Component';
import { Headline } from 'src/helpers/Headline';
import { LinkWrapper } from 'src/helpers/LinkWrapper';
import ImageWrapper from 'src/helpers/Media/ImageWrapper';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';
import { getEnum } from 'src/lib/utils';
import { GenericChartTheme, GetTextAlignment } from './GenericChart.theme';

export type GenericChartProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.General.GenericChart.GenericChart & {
    fields: {
      children: [
        Feature.EnterpriseWeb.AndersenWindows.Components.General.GenericChart.GenericChartColumn
      ];
    };
  };

export type GenericChartColumnProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.General.GenericChart.GenericChartColumn;

const RenderTitleContent = (column: GenericChartColumnProps) => {
  if (!column?.fields) {
    return <></>;
  }

  return (
    <>
      {!column.fields.titleLink?.value?.href ? (
        column.fields.title?.value && (
          <h3>
            <Text field={column.fields.title} encode={false}></Text>
          </h3>
        )
      ) : (
        <LinkWrapper
          field={column.fields.titleLink}
          ariaLabel={{ value: column.fields.title?.value || 'Title Link' }}
        >
          {column.fields.title?.value ? (
            <h3>
              <span className="border-black hover:border-b-2">
                <Text field={column.fields.title} encode={false}></Text>
              </span>
            </h3>
          ) : (
            <h3>{column.fields.titleLink.value?.text}</h3>
          )}
        </LinkWrapper>
      )}
      {!column.fields.subtitleLink?.value?.href ? (
        column.fields.subtitle?.value && (
          <h4>
            <Text field={column.fields.subtitle}></Text>
          </h4>
        )
      ) : (
        <LinkWrapper
          field={column.fields.subtitleLink}
          ariaLabel={{ value: column.fields.subtitle?.value || 'Subtitle Link' }}
        >
          {column.fields.subtitle?.value ? (
            <h4>
              <span className="border-black hover:border-b-2">
                <Text field={column.fields.subtitle} encode={false}></Text>
              </span>
            </h4>
          ) : (
            <h4>{column.fields.subtitleLink.value?.text}</h4>
          )}
        </LinkWrapper>
      )}
    </>
  );
};

export type TextAlignment = 'left' | 'center' | 'right' | '';
export type BackgroundColor = 'black' | 'gray' | 'white';

const GenericChart = (props: GenericChartProps) => {
  const alignment = getEnum<TextAlignment>(props.fields?.textAlignment) || 'left';
  const titleRowBackgroundColor =
    getEnum<BackgroundColor>(props.fields?.titleRowBackgroundColor) || 'black';
  const titleColumnBackgroundColor =
    getEnum<BackgroundColor>(props.fields?.titleColumnBackgroundColor) || 'gray';
  const { themeData } = useTheme(
    GenericChartTheme(titleRowBackgroundColor, titleColumnBackgroundColor, alignment)
  );
  const MAX_CHART_ROWS = 25;
  const countToMaxChartRowsArray = [...Array(MAX_CHART_ROWS + 1).keys()].splice(1);

  const hasSchemaText = !!props.fields.schemaName?.value && !!props.fields.schemaDescription?.value;
  const hasImageRow = !!props.fields.children.find(
    (_: Feature.EnterpriseWeb.AndersenWindows.Components.General.GenericChart.GenericChartColumn) =>
      !!_.fields?.image?.value?.src
  );
  const hasImageColumn = !!countToMaxChartRowsArray.find(
    (count: number) => !!props.fields[`row${count}Image`]?.value?.src
  );

  return (
    <Component
      sectionWrapperClasses={themeData.classes.section}
      variant="lg"
      grid=" w-full "
      dataComponent="general/genericchart"
      {...props}
    >
      <Headline
        {...props}
        classes={' text-[26px] font-demi leading-[1.25] mb-[12px] md:text-[34px] '}
      ></Headline>
      <BodyCopy
        {...props}
        classes={' hidden md:block text-[18px] leading-1.5] mb-[10px] '}
      ></BodyCopy>
      <div className="comparison-chart-table">
        {hasSchemaText && (
          <>
            <meta itemProp="name" content={props.fields.schemaName?.value} />
            <meta itemProp="description" content={props.fields.schemaDescription?.value} />
          </>
        )}

        <table className={themeData.classes.table}>
          <thead>
            <tr className={themeData.classes.theadRow}>
              {hasImageColumn && <th className={themeData.classes.imageColumnHeader}>&nbsp;</th>}
              <th>&nbsp;</th>
              {props.fields.children.map((column: GenericChartColumnProps, index: number) => (
                <th
                  className={themeData.classes.rowHeader}
                  key={'GenericChartColumnProps_th_' + index}
                >
                  {RenderTitleContent(column)}
                </th>
              ))}
            </tr>
          </thead>
          <tbody>
            {hasImageRow && (
              <tr className={themeData.classes.imageRow}>
                {hasImageColumn && <td className={themeData.classes.imageColumn}>&nbsp;</td>}
                <th>&nbsp;</th>
                {props.fields.children.map((column: GenericChartColumnProps, index: number) => (
                  <td key={'GenericChartColumnProps_td_' + index} className="p-[5px]">
                    {!column?.fields?.imageLink?.value?.href ? (
                      <ImageWrapper
                        image={column?.fields?.image}
                        additionalDesktopClasses="lazy"
                        imageLayout="intrinsic"
                      ></ImageWrapper>
                    ) : (
                      <LinkWrapper
                        field={column?.fields?.imageLink}
                        ariaLabel={{ value: 'GenericChart image' }}
                      >
                        <ImageWrapper
                          image={column?.fields?.image}
                          additionalDesktopClasses="lazy"
                          imageLayout="intrinsic"
                        ></ImageWrapper>
                      </LinkWrapper>
                    )}
                  </td>
                ))}
              </tr>
            )}
            {countToMaxChartRowsArray.map((rowNumber: number, index: number) => {
              const rowTitle = props.fields[`row${rowNumber}Title`] as TextField;
              const rowTitleLink = props.fields[`row${rowNumber}Link`] as LinkField;
              const rowSubtitle = props.fields[`row${rowNumber}Subtitle`] as TextField;
              const rowSubtitleLink = props.fields[`row${rowNumber}SubtitleLink`] as LinkField;
              const rowImage = props.fields[`row${rowNumber}Image`] as ImageField;
              const rowImageLink = props.fields[`row${rowNumber}ImageLink`] as LinkField;
              return (
                (!!rowTitle?.value || !!rowTitleLink?.value?.href) && (
                  <tr className={themeData.classes.tableRow} key={'content-row-' + index}>
                    {hasImageColumn && (
                      <td className={themeData.classes.imageColumn}>
                        {rowImage?.value?.src &&
                          (!rowImageLink?.value?.href ? (
                            <>
                              <ImageWrapper image={rowImage}></ImageWrapper>
                            </>
                          ) : (
                            <>
                              <LinkWrapper field={rowImageLink} ariaLabel={{ value: 'Row Image' }}>
                                <ImageWrapper image={rowImage}></ImageWrapper>
                              </LinkWrapper>
                            </>
                          ))}
                      </td>
                    )}
                    <th className={themeData.classes.columnHeader}>
                      {!rowTitleLink?.value?.href ? (
                        rowTitle?.value && (
                          <h4>
                            <Text field={rowTitle} encode={false}></Text>
                          </h4>
                        )
                      ) : (
                        <LinkWrapper
                          field={rowTitleLink}
                          ariaLabel={{
                            value: (rowTitleLink?.value?.text as string) || 'Row Title Link',
                          }}
                        >
                          {rowTitle?.value ? (
                            <h4>
                              <Text field={rowTitle} encode={false}></Text>
                            </h4>
                          ) : (
                            <h4>{rowTitleLink?.value?.text}</h4>
                          )}
                        </LinkWrapper>
                      )}
                      {!rowSubtitleLink?.value?.href ? (
                        rowSubtitle?.value && (
                          <h5>
                            <Text field={rowSubtitle} encode={false}></Text>
                          </h5>
                        )
                      ) : (
                        <LinkWrapper
                          field={rowSubtitleLink}
                          ariaLabel={{
                            value: (rowSubtitleLink?.value?.text as string) || 'Row Subtitle Link',
                          }}
                        >
                          {rowSubtitle?.value ? (
                            <h5>
                              <Text field={rowSubtitle} encode={false}></Text>
                            </h5>
                          ) : (
                            <h5>{rowSubtitleLink?.value?.text}</h5>
                          )}
                        </LinkWrapper>
                      )}
                    </th>
                    {props.fields.children.map(
                      (
                        column: Feature.EnterpriseWeb.AndersenWindows.Components.General.GenericChart.GenericChartColumn,
                        index: number
                      ) => {
                        const textAlignment =
                          column.fields?.textAlignment != null
                            ? getEnum<TextAlignment>(column.fields.textAlignment) || ''
                            : '';
                        const rowContent = (column?.fields?.[
                          `row${rowNumber}Content` as keyof typeof column.fields
                        ] || { value: '' }) as RichTextField;
                        return (
                          <td
                            className={`${themeData.classes.tableData} ${GetTextAlignment(
                              textAlignment
                            )}`}
                            key={'content_td_' + index}
                          >
                            <div className={themeData.classes.mobileTitleColumn}>
                              {RenderTitleContent(column)}
                            </div>
                            <RichTextWrapper
                              field={rowContent}
                              classes={themeData.classes.contentClass}
                            ></RichTextWrapper>
                          </td>
                        );
                      }
                    )}
                  </tr>
                )
              );
            })}
          </tbody>
        </table>
      </div>
      {props.fields.footer?.value && (
        <div className={themeData.classes.footer}>
          <RichTextWrapper field={props.fields.footer}></RichTextWrapper>
        </div>
      )}
    </Component>
  );
};

export default withDatasourceCheck()<GenericChartProps>(GenericChart);
