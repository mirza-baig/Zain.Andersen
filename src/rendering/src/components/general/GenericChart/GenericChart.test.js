// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import GenericChart from './GenericChart';
import defaultData from './GenericChart.mock-data';

it('renders correctly', async () => {
  const component = snapshot(GenericChart, { componentProps: defaultData });
  await hasDataComponent(component, 'general/genericchart');
});
