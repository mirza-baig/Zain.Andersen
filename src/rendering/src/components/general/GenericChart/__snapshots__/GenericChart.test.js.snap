// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  class="font-futura-pt"
  data-component="general/genericchart"
  id="id964553899"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class=" w-full  px-m gap-s md:gap-s "
    >
      <div
        class="items-top font-sans [&_a:hover]:underline  text-[26px] font-demi leading-[1.25] mb-[12px] md:text-[34px] "
      >
        <h2>
          Grille Types
        </h2>
      </div>
      <div
        class=" hidden md:block text-[18px] leading-1.5] mb-[10px] "
      >
        <div
          class="body-copy"
        >
          <div>
            <p>
              Module body copy text here
            </p>
          </div>
        </div>
      </div>
      <div
        class="comparison-chart-table"
      >
        <table
          class=" w-full table-fixed text-center"
        >
          <thead>
            <tr
              class=" title-row hidden md:table-row "
            >
              <th>
                 
              </th>
              <th
                class="row-header p-[5px] bg-gray text-black leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h3]:mb-[10px] [&_h4]:text-[18px] [&_h4]:font-demi [&_h4]:mb-[8px]"
              >
                <h3>
                  <span>
                    Profile Width
                  </span>
                </h3>
                <h4>
                  Subtitle
                </h4>
              </th>
              <th
                class="row-header p-[5px] bg-gray text-black leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h3]:mb-[10px] [&_h4]:text-[18px] [&_h4]:font-demi [&_h4]:mb-[8px]"
              >
                <h3>
                  <span>
                    Profile Picture
                  </span>
                </h3>
                <h4>
                  subtitle 2
                </h4>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class=" image-row hidden md:table-row"
            >
              <th>
                 
              </th>
              <td
                class="p-[5px]"
              />
              <td
                class="p-[5px]"
              >
                <div
                  class="relative hidden md:block   lazy"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27402%27%20height=%27402%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="100 Series Patio Door"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
                <div
                  class="relative w-full md:hidden"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27402%27%20height=%27402%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="100 Series Patio Door"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                    />
                    <noscript />
                  </span>
                </div>
              </td>
            </tr>
            <tr
              class=" content-row "
            >
              <th
                class="column-header border border-black bg-black text-white border-b-gray border-b text-[14px] md:text-[18px] p-[5px_15px] font-demi"
              >
                <h4>
                  <span>
                    Ovolo
                  </span>
                </h4>
                
              </th>
              <td
                class="bg-white border border-black block md:table-cell border-b-black "
              >
                <div
                  class=" mobile-title md:hidden p-[5px] bg-gray text-black border-b-black border-b leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h4]:text-[14px] [&_h4]:font-demi"
                >
                  <h3>
                    <span>
                      Profile Width
                    </span>
                  </h3>
                  <h4>
                    Subtitle
                  </h4>
                </div>
                <div
                  class="body-copy  [&_img]:mx-auto p-[5px]"
                >
                  <div>
                    7/8” & 1 
                    <awsup>
                      1/4"
                    </awsup>
                    <br />
                    

                    <div>
                       
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="bg-white border border-black block md:table-cell border-b-black "
              >
                <div
                  class=" mobile-title md:hidden p-[5px] bg-gray text-black border-b-black border-b leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h4]:text-[14px] [&_h4]:font-demi"
                >
                  <h3>
                    <span>
                      Profile Picture
                    </span>
                  </h3>
                  <h4>
                    subtitle 2
                  </h4>
                </div>
                <div
                  class="body-copy  [&_img]:mx-auto p-[5px]"
                >
                  <div>
                    <p
                      style="text-align: justify;"
                    >
                      <img
                        alt="ovolo grille profiles"
                        height="215"
                        src="https://preview.local.andersenwindows.com/-/media/AndersenWindows/images/windows-and-doors/options-and-accessories/grilles-new/profiles/aw-ovolo.jpg?rev=a6c8d20d1d71442b92db2cf4d7ce822f&hash=D5BE19BB9E5F91D033AB77A106253A3F"
                        width="272"
                      />
                    </p>
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class=" content-row "
            >
              <th
                class="column-header border border-black bg-black text-white border-b-gray border-b text-[14px] md:text-[18px] p-[5px_15px] font-demi"
              >
                <h4>
                  <span>
                    Contemporary
                  </span>
                </h4>
                
              </th>
              <td
                class="bg-white border border-black block md:table-cell border-b-black "
              >
                <div
                  class=" mobile-title md:hidden p-[5px] bg-gray text-black border-b-black border-b leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h4]:text-[14px] [&_h4]:font-demi"
                >
                  <h3>
                    <span>
                      Profile Width
                    </span>
                  </h3>
                  <h4>
                    Subtitle
                  </h4>
                </div>
                <div
                  class="body-copy  [&_img]:mx-auto p-[5px]"
                >
                  <div>
                    1 
                    <awsup>
                      1/4"
                    </awsup>
                    <br />
                    

                    <div>
                       
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="bg-white border border-black block md:table-cell border-b-black "
              >
                <div
                  class=" mobile-title md:hidden p-[5px] bg-gray text-black border-b-black border-b leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h4]:text-[14px] [&_h4]:font-demi"
                >
                  <h3>
                    <span>
                      Profile Picture
                    </span>
                  </h3>
                  <h4>
                    subtitle 2
                  </h4>
                </div>
                <div
                  class="body-copy  [&_img]:mx-auto p-[5px]"
                >
                  <div>
                    <img
                      alt="contemp grille profile"
                      height="215"
                      src="https://preview.local.andersenwindows.com/-/media/AndersenWindows/images/windows-and-doors/options-and-accessories/grilles-new/profiles/aw-contemp.jpg?rev=afc94be4f9d949e5bde4c59171e90dae&hash=6CD2E908B43D8917178FE3BCC1F360A9"
                      width="272"
                    />
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class=" content-row "
            >
              <th
                class="column-header border border-black bg-black text-white border-b-gray border-b text-[14px] md:text-[18px] p-[5px_15px] font-demi"
              >
                <h4>
                  <span>
                    E-Series Contemporary Full Divided Light
                  </span>
                </h4>
                
              </th>
              <td
                class="bg-white border border-black block md:table-cell border-b-black "
              >
                <div
                  class=" mobile-title md:hidden p-[5px] bg-gray text-black border-b-black border-b leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h4]:text-[14px] [&_h4]:font-demi"
                >
                  <h3>
                    <span>
                      Profile Width
                    </span>
                  </h3>
                  <h4>
                    Subtitle
                  </h4>
                </div>
                <div
                  class="body-copy  [&_img]:mx-auto p-[5px]"
                >
                  <div>
                    7/8” 
                    <br />
                    

                    <div>
                       
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="bg-white border border-black block md:table-cell border-b-black "
              >
                <div
                  class=" mobile-title md:hidden p-[5px] bg-gray text-black border-b-black border-b leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h4]:text-[14px] [&_h4]:font-demi"
                >
                  <h3>
                    <span>
                      Profile Picture
                    </span>
                  </h3>
                  <h4>
                    subtitle 2
                  </h4>
                </div>
                <div
                  class="body-copy  [&_img]:mx-auto p-[5px]"
                >
                  <div>
                    <img
                      alt="e-series grille profile"
                      height="215"
                      src="https://preview.local.andersenwindows.com/-/media/AndersenWindows/images/windows-and-doors/options-and-accessories/grilles-new/profiles/aw-e-ser.jpg?rev=cb464568eb6a44f58f9df792f8b6b3f0&hash=03080AB03D0728798A3E068EDF9B33BE"
                      width="272"
                    />
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class=" content-row "
            >
              <th
                class="column-header border border-black bg-black text-white border-b-gray border-b text-[14px] md:text-[18px] p-[5px_15px] font-demi"
              >
                <h4>
                  <span>
                    Squared
                  </span>
                </h4>
                
              </th>
              <td
                class="bg-white border border-black block md:table-cell border-b-black "
              >
                <div
                  class=" mobile-title md:hidden p-[5px] bg-gray text-black border-b-black border-b leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h4]:text-[14px] [&_h4]:font-demi"
                >
                  <h3>
                    <span>
                      Profile Width
                    </span>
                  </h3>
                  <h4>
                    Subtitle
                  </h4>
                </div>
                <div
                  class="body-copy  [&_img]:mx-auto p-[5px]"
                >
                  <div>
                    1", 1 
                    <awsup>
                      1/4"
                    </awsup>
                    , 1 
                    <awsup>
                      1/2"
                    </awsup>
                    , 1 
                    <awsup>
                      3/4"
                    </awsup>
                    , 2", 
                    <br />
                    
2 
                    <awsup>
                      1/4"
                    </awsup>
                    , 2 
                    <awsup>
                      1/2"
                    </awsup>
                    , 3" & 4"
                  </div>
                </div>
              </td>
              <td
                class="bg-white border border-black block md:table-cell border-b-black "
              >
                <div
                  class=" mobile-title md:hidden p-[5px] bg-gray text-black border-b-black border-b leading-[1.25] [&_h3]:text-[22px] [&_h3]:font-bold [&_h4]:text-[14px] [&_h4]:font-demi"
                >
                  <h3>
                    <span>
                      Profile Picture
                    </span>
                  </h3>
                  <h4>
                    subtitle 2
                  </h4>
                </div>
                <div
                  class="body-copy  [&_img]:mx-auto p-[5px]"
                >
                  <div>
                    <img
                      alt="square grille profile"
                      height="215"
                      src="https://preview.local.andersenwindows.com/-/media/AndersenWindows/images/windows-and-doors/options-and-accessories/grilles-new/profiles/aw-square.jpg?rev=0c099518ea9941be9e7879f90bf2aa42&hash=2DDF23D91CADB83E6C0FBB48AB2DDACE"
                      width="272"
                    />
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="content-footer"
      >
        <div
          class="body-copy"
        >
          <div>
            <p>
              Footer copy
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
